src/features/content-explorer/item-list/ItemListIcon.js

```diff
@@ -1,40 +1,12 @@
-import PropTypes from 'prop-types';
 import React from 'react';
-import { FormattedMessage } from 'react-intl';
-
-import BookmarkIcon from '../../../icons/bookmark-icon';
-import FileIcon from '../../../icons/file-icon';
-import FolderIcon from '../../../icons/folder-icon';
+import PropTypes from 'prop-types';
 
-import ItemTypes from '../item-types';
 import { ItemTypePropType } from '../prop-types';
-import messages from '../messages';
+import IconCell from '../../../elements/common/item/IconCell';
 
-const ItemListIcon = ({ type, extension = '', hasCollaborations = false, isExternallyOwned = false }) => {
-    switch (type) {
-        case ItemTypes.FOLDER: {
-            let titleID = 'personalFolder';
-            if (hasCollaborations) {
-                titleID = 'collaboratedFolder';
-            } else if (isExternallyOwned) {
-                titleID = 'externalFolder';
-            }
-            return (
-                <FolderIcon
-                    isCollab={hasCollaborations}
-                    isExternal={isExternallyOwned}
-                    title={<FormattedMessage {...messages[titleID]} />}
-                />
-            );
-        }
-        case ItemTypes.FILE:
-            return <FileIcon extension={extension} title={<FormattedMessage {...messages.file} />} />;
-        case ItemTypes.BOOKMARK:
-            return <BookmarkIcon title={<FormattedMessage {...messages.bookmark} />} />;
-        default:
-            // Use generic file icon as fallback
-            return <FileIcon title={<FormattedMessage {...messages.file} />} />;
-    }
+const ItemListIcon = ({ type, extension, hasCollaborations = false, isExternallyOwned = false }) => {
```

Comment:
Can we directly replace ItemListIcon with IconCell in ItemList?

Reply:
I double checked and it doesn't seem like it's being used internally, so I guess we could. But I could also see us adding props to `IconCell` so that it's more versatile (like how it's done in EUA).

Reply:
That makes sense 👍